proposals/lip-0067.md

```diff
@@ -83,36 +80,38 @@ In the following sections, we describe the uses of the properties of `keystoreSc
 
-##### ciphertext
+The specified function encrypts the message using the secret key; to decrypt it, the secret key along with `cipher` and `cipherparams` must be used. If the secret key is longer than the key size required by the encoding function, it is truncated to the correct number of bits. The following option is supported:
 
-The encrypted message in hexadecimal format.
+| `cipher`      | function      | `cipherparams` | Definition |
+|---------------|---------------|----------------|------------|
+| "AES-128-GCM" | `aes-128-gcm` | `{iv: string}` | [RFC 7714](https://datatracker.ietf.org/doc/html/rfc7714) |
 
-##### mac
+##### ciphertext
 
-Computed as `SHA256(encryptionKey[-16:] + cipherText)`, where `encryptionKey` is the key derived from the user password using the key-derivation function. The `mac` can be used to verify the encryption key before starting the decryption process.
+The encrypted message in hexadecimal format. It is generated by inputting the message and the secret key to `aes-128-gcm` function.
 
 ##### kdf and kdfparams
 
-The encryption key is an intermediate key derived from the user password. It is used to generate the secret key for decryption, and verify if the given password is correct. The function, and the params used to derive this key from the password are specified in `kdf`. The following values of `kdf` and `kdfparams` are allowed, depending on the key-derivation function:
+The `derivedKey` is an intermediate key derived from the user password: a derivation function is applied to the user password. The derived key is used to generate the secret key for decryption, and verify if the given password is correct. The secret key is the leftmost 16 bytes of the derived key. The function, and the params used to get the derivation key from the password are specified in `kdf`. The following values of `kdf` and `kdfparams` are allowed, depending on the key-derivation function:
 
 | `kdf`            | function   | `kdfparams` | Definition |
 |------------------|------------|-------------|------------|
 | "PBKDF2-SHA-256" | `pbkdf2`   | `{iterations: uint32, salt: string}` | [RFC 2898](https://www.ietf.org/rfc/rfc2898.txt) |
 | "argon2id"       | `argon2id` | `{parallelism: uint32, iterations: uint32, memory: uint32, salt: string}` | [RFC 9106](https://datatracker.ietf.org/doc/html/rfc9106) |
 
-##### cipher and cipherparams
+##### mac
 
-The specified function encrypts the secret using the encryption key; to decrypt it, the encryption key along with `cipher` and `cipherparams` must be used. If the encryption key is longer than the key size required by the encoding function, it is truncated to the correct number of bits. The following option is supported:
+Computed as `SHA256(derivedKey[-16:] + ciphertext)`, where `derivedKey` is the intermediate key derived from the user password using the key-derivation function. The `mac` can be used to verify the secret key before starting the decryption process.
```

Comment:
> The `mac` can be used to verify the secret key before starting the decryption process.

Especially here it can be confusing what is meant with "secret key". Because it is used to verify secret key of the signature scheme a.k.a the decrypted message. But the reader might think it is about the encryption key.

Reply:
I agree with @AndreasKendziorra having a point, so after all the term `encryption key` is better than `secret key`, especially since it can be also used in the context of decryption. 
Also, note that in the first sentence you explain `where derivedKey is the intermediate key derived from the user password using the key-derivation function`, which is maybe unnecessary since you already explain what `derivedKey` is in the `kdf and kdfparams` subsection.

Reply:
> > The `mac` can be used to verify the secret key before starting the decryption process.
> 
> Especially here it can be confusing what is meant with "secret key". Because it is used to verify secret key of the signature scheme a.k.a the decrypted message. But the reader might think it is about the encryption key.

@AndreasKendziorra I have updated this line in LIP but you can check once if it is correctly written.

Reply:
I was actually wrong. This mac is primarily used to verify that the password for the key derivation function was correct. So, when `derivedKey` is derived, it should be verified with `mac` that the value is correct. If not, either the password was wrong, or the ciphertext was altered. But altering the the ciphertext in this JSON is probably not a realistic scenario. Therefore, I would write:

"Computed as `SHA256(derivedKey[-16:] + ciphertext)`. The `mac` should be used to verify that the password for the key derivation function was correct. This is done by computing the `mac` based on the derived key and ciphertext. If this `mac` does not match with the one in the JSON file the password was not correct."

Reply:
Done